File: Data Structures & Algorithms/same-binary-tree/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    bool sameTree=true;
public:
    void traverse(TreeNode* pNode,TreeNode* qNode) {
        if(pNode->val!=qNode->val)sameTree=false;
        if(pNode->left==nullptr && qNode->left!=nullptr)sameTree=false;
        if(pNode->right==nullptr && qNode->right!=nullptr)sameTree=false;

        if(sameTree) {
            if(pNode->left !=nullptr && qNode->left!=nullptr)traverse(pNode->left,qNode->left);
            if(pNode->right!=nullptr && qNode->right!=nullptr)traverse(pNode->right,qNode->right);
        }
    }

    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(p==nullptr && q==nullptr)return true;
        if(p==nullptr)return false;
        if(q==nullptr)return false;

        traverse(p,q);
        return sameTree;
    }
};

File: Data Structures & Algorithms/valid-binary-search-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        in_order_traversal=[]
        def inOrderTraversal(node):
            nonlocal in_order_traversal
            if node.left:
                inOrderTraversal(node.left)
            in_order_traversal.append(node.val)
            #print(node.val)
            if  node.right:
                inOrderTraversal(node.right)
            
        def isSorted(li):
            for i in range(1,len(li)):
                if li[i]<=li[i-1]:return False
            return True
        
        inOrderTraversal(root)
        #print(in_order_traversal)
        return isSorted(in_order_traversal)